multiple matching crates for `rustc_serialize` panic

### Minimal repro:

```rust
#![feature(rustc_private)]
extern crate rustc_serialize;
use rustc_serialize::hex::ToHex;

fn main() {
    println!("Use toHex: {}", [0xFF_u8].to_hex());
}
```

 - rustc:   `nightly-x86_64-apple-darwin - rustc 1.17.0-nightly (49c67bd63 2017-03-24)`
 - clippy `v0.0.121`


### Error:

```rust
➜ cargo clippy
   Compiling clippy_panic v0.1.0 (file:///Users/gib/tmp/clippy_panic)
error[E0464]: multiple matching crates for `rustc_serialize`
 --> src/main.rs:2:1
  |
2 | extern crate rustc_serialize;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: candidates:
  = note: path: /Users/gib/tmp/clippy_panic/target/debug/deps/librustc_serialize-54d2bf33b1d4b0b0.rlib
  = note: crate name: rustc_serialize
  = note: path: /Users/gib/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/librustc_serialize-fde13340fc7641e7.rlib
  = note: crate name: rustc_serialize

error[E0463]: can't find crate for `rustc_serialize`
 --> src/main.rs:2:1
  |
2 | extern crate rustc_serialize;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't find crate

error: aborting due to 2 previous errors

thread 'rustc' panicked at 'Box<Any>', src/librustc_errors/lib.rs:455
note: Run with `RUST_BACKTRACE=1` for a backtrace.
thread 'main' panicked at 'rustc_thread failed: Any', src/libcore/result.rs:859
error: Could not compile `clippy_panic`.

To learn more, run the command again with --verbose.
  = note: path: /Users/gib/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/librustc_serialize-fde13340fc7641e7.rlib
  = note: crate name: rustc_serialize

error[E0463]: can't find crate for `rustc_serialize`
 --> src/main.rs:3:1
  |
3 | extern crate rustc_serialize;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't find crate

error: aborting due to 2 previous errors

thread 'rustc' panicked at 'Box<Any>', src/librustc_errors/lib.rs:455
note: Run with `RUST_BACKTRACE=1` for a backtrace.
thread 'main' panicked at 'rustc_thread failed: Any', src/libcore/result.rs:859
error: Could not compile `five`.

To learn more, run the command again with --verbose.
```

Okay, that works for my (I guess too-minimal) test case, but not for the app I originally had the problem with, as that one also uses the `crypto` crate (looks like that depends on the external `rustc-serialize`).

#### Cargo.toml:

```toml
[dependencies]
rust-crypto = "^0.2"
```

#### Cargo.lock (relevant lines)

```toml
[[package]]
name = "rust-crypto"
version = "0.2.36"
source = "registry+https://github.com/rust-lang/crates.io-index"
dependencies = [
 "gcc 0.3.45 (registry+https://github.com/rust-lang/crates.io-index)",
 "libc 0.2.21 (registry+https://github.com/rust-lang/crates.io-index)",
 "rand 0.3.15 (registry+https://github.com/rust-lang/crates.io-index)",
 "rustc-serialize 0.3.23 (registry+https://github.com/rust-lang/crates.io-index)",
 "time 0.1.36 (registry+https://github.com/rust-lang/crates.io-index)",
]

[[package]]
name = "rustc-serialize"
version = "0.3.23"
source = "registry+https://github.com/rust-lang/crates.io-index"
```

#### Output

```rust
➜ rm -rf target && cargo clippy                                                                                                                                            ~/tmp/clippy_panic
   Compiling rustc-serialize v0.3.23
   Compiling gcc v0.3.45
   Compiling libc v0.2.21
   Compiling time v0.1.36
   Compiling rand v0.3.15
   Compiling rust-crypto v0.2.36
   Compiling five v0.1.0 (file:///Users/gib/tmp/clippy_panic)
error[E0464]: multiple matching crates for `rustc_serialize`
 --> src/main.rs:2:1
  |
2 | extern crate rustc_serialize;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: candidates:
  = note: path: /Users/gib/tmp/clippy_panic/target/debug/deps/librustc_serialize-54d2bf33b1d4b0b0.rlib
  = note: crate name: rustc_serialize
  = note: path: /Users/gib/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/librustc_serialize-fde13340fc7641e7.rlib
  = note: crate name: rustc_serialize

error[E0463]: can't find crate for `rustc_serialize`
 --> src/main.rs:2:1
  |
2 | extern crate rustc_serialize;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't find crate

error: aborting due to 2 previous errors

thread 'rustc' panicked at 'Box<Any>', src/librustc_errors/lib.rs:455
note: Run with `RUST_BACKTRACE=1` for a backtrace.
thread 'main' panicked at 'rustc_thread failed: Any', src/libcore/result.rs:859
error: Could not compile `five`.

To learn more, run the command again with --verbose.
```

--- a/src/main.rs
+++ b/src/main.rs
@@ -169,8 +169,6 @@ pub fn main() {
         return;
     }
 
-    let dep_path = env::current_dir().expect("current dir is not readable").join("target").join("debug").join("deps");
-
     if let Some("clippy") = std::env::args().nth(1).as_ref().map(AsRef::as_ref) {
         // this arm is executed on the initial call to `cargo clippy`
 
@@ -207,12 +205,11 @@ pub fn main() {
             let args = std::env::args().skip(2);
             if let Some(first) = target.kind.get(0) {
                 if target.kind.len() > 1 || first.ends_with("lib") {
-                    if let Err(code) = process(std::iter::once("--lib".to_owned()).chain(args), &dep_path) {
+                    if let Err(code) = process(std::iter::once("--lib".to_owned()).chain(args)) {
                         std::process::exit(code);
                     }
                 } else if ["bin", "example", "test", "bench"].contains(&&**first) {
-                    if let Err(code) = process(vec![format!("--{}", first), target.name].into_iter().chain(args),
-                                               &dep_path) {
+                    if let Err(code) = process(vec![format!("--{}", first), target.name].into_iter().chain(args)) {
                         std::process::exit(code);
                     }
                 }
@@ -271,9 +268,8 @@ pub fn main() {
     }
 }
 
-fn process<P, I>(old_args: I, dep_path: P) -> Result<(), i32>
-    where P: AsRef<Path>,
-          I: Iterator<Item = String>
+fn process<I>(old_args: I) -> Result<(), i32>
+    where I: Iterator<Item = String>
 {
 
     let mut args = vec!["rustc".to_owned()];
@@ -286,8 +282,6 @@ fn process<P, I>(old_args: I, dep_path: P) -> Result<(), i32>
     if !found_dashes {
         args.push("--".to_owned());
     }
-    args.push("-L".to_owned());
-    args.push(dep_path.as_ref().to_string_lossy().into_owned());
     args.push("-Zno-trans".to_owned());
     args.push("--cfg".to_owned());
     args.push(r#"feature="cargo-clippy""#.to_owned());

